Foundation/FileManager.swift

```diff
@@ -32,10 +32,64 @@ open class FileManager : NSObject {
         }
     }
     
-    /* Returns an NSArray of NSURLs locating the mounted volumes available on the computer. The property keys that can be requested are available in NSURL.
-     */
+    /// Returns an array of URLs that identify the mounted volumes available on the device.
     open func mountedVolumeURLs(includingResourceValuesForKeys propertyKeys: [URLResourceKey]?, options: VolumeEnumerationOptions = []) -> [URL]? {
-        NSUnimplemented()
+        var urls: [URL]
+
+#if os(Linux)
+        guard let procMounts = try? String(contentsOfFile: "/proc/mounts", encoding: .utf8) else {
+            return nil
+        }
+        urls = []
+        for line in procMounts.components(separatedBy: "\n") {
+            let mountPoint = line.components(separatedBy: " ")
+            if mountPoint.count > 2 {
+                urls.append(URL(fileURLWithPath: mountPoint[1], isDirectory: true))
+            }
+        }
+#elseif os(macOS)
```

Comment:
Can you use canImport(Darwin) here instead? I'm ok with os(Linux) for the above because /proc/mounts is explicitly a Linux feature, but I'd rather not tie one Darwin API to a specific OS.

Reply:
I'm also half-thinking whether we also need to start using:

```swift
#else
#error("Requires a platform-specific implementation")
#endif
```

now that we _do_ have `#error`.